tests/end-to-end/upgrade/pkg/tests/function/function.go

```
@@ -0,0 +1,277 @@
+package function
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	kubelessV1 "github.com/kubeless/kubeless/pkg/apis/kubeless/v1beta1"
+	kubeless "github.com/kubeless/kubeless/pkg/client/clientset/versioned"
+	kymaApi "github.com/kyma-project/kyma/components/api-controller/pkg/apis/gateway.kyma-project.io/v1alpha2"
+	kyma "github.com/kyma-project/kyma/components/api-controller/pkg/clients/gateway.kyma-project.io/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	extensionsv1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	instr "k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+)
+
+// LambdaFunctionUpgradeTest tests the creation of a kubeless function and execute a http request to the exposed api of the function after Kyma upgrade phase
+type LambdaFunctionUpgradeTest struct {
+	functionName, uuid string
+	kubelessClient     *kubeless.Clientset
+	coreClient         *kubernetes.Clientset
+	apiClient          *kyma.Clientset
+	nSpace             string
+	hostName           string
+}
+
+func int32Ptr(i int32) *int32 { return &i }
+
+// NewLambdaFunctionUpgradeTest returns new instance of the FunctionUpgradeTest
+func NewLambdaFunctionUpgradeTest(config *restclient.Config) *LambdaFunctionUpgradeTest {
+
+	kubelessClient, err := kubeless.NewForConfig(config)
+	if err != nil {
+		return &LambdaFunctionUpgradeTest{}
```

this hides an error, is it really what we want?